packages/evolution-frontend/src/components/inputs/InputMapFindPlace.tsx

```diff
@@ -228,6 +263,11 @@ export class InputMapFindPlace<CustomSurvey, CustomHousehold, CustomHome, Custom
             selectedPlace.properties = { ...selectedPlaceProperties };
             selectedPlace.properties.lastAction = 'findPlace';
             selectedPlace.properties.geocodingQueryString = this.state.geocodingQueryString;
+            selectedPlace.properties.geocodingResultsData = {
+                formatted_address: placeData.formatted_address,
+                place_id: placeData.place_id,
```

Comment:
Are we allowed to save this? But this is the main InputMapFindPlace class, so not google specific. If we are not allowed, this is not where we should take care of this, but make sure that in the present case with google, we respect the license

Reply:
https://cloud.google.com/maps-platform/terms/maps-service-terms

> 5. Places API
> 
> The following terms apply only to the Places API:
> ...
> 5.4 Caching. Customer can temporarily cache **latitude (lat) and longitude (lng) values from the Places API for up to 30 consecutive calendar days**, after which Customer must delete the cached latitude and longitude values. Customer can cache Places API Place ID (place_id) values, in accordance with the [Places API Policies](https://developers.google.com/maps/documentation/places/web-service/policies).

https://developers.google.com/maps/documentation/places/web-service/policies

> Note that **the place ID, used to uniquely identify a place, is exempt from the caching restriction.** You can therefore store place ID values indefinitely. The place ID is returned in the place_id field in Places API responses.

https://cloud.google.com/maps-platform/terms
> 
> 3.2.3 Restrictions Against Misusing the Services.
> 
> (a)  No Scraping. Customer will not export, extract, or otherwise scrape Google Maps Content for use outside the Services. For example, Customer will not: (i) pre-fetch, index, store, reshare, or rehost Google Maps Content outside the services; (ii) bulk download Google Maps tiles, Street View images, geocodes, directions, distance matrix results, roads information, places information, elevation values, and time zone details; **(iii) copy and save business names, addresses, or user reviews;** or (iv) use Google Maps Content with text-to-speech services.

Donc, si j'ai bien compris:
1. Place ID: OK ✅ 
2. Lat Lon: 30 jours seulement? ❓ Est-ce que ceci pourrait être un problème pour l'enquête OD en général?
3. address: Non ❌ , donc on peut pas enregistrer le type d'adresse non plus.

cc @kaligrafy

Reply:
On viole la license depuis le début, comme tout le monde. Pour régler ça, faudra vraiment que tous les partenaires financent des ressources pour OpenStreetMap au plus vite. Je ne fais plus d'enquête avec Google à l'avenir!

Reply:
Ils ne peuvent pas protéger l'adresse, c'est public, ce n'est pas protégé. Le lat lon associé par Google à l'adresse, lui est protégé. Pour le type de précision, c'est ok si on l'utilise live pour aider l'usager, mais effectivement, pas sur qu'on peut garder ça longtemps et faire quelque chose avec... Mais bon, après on est académique et gouvernmental et il y a le fameux "fair-use" qui te permet de faire ce que tu veux pour le greater good si ce n'est pas commercial, mais ça se discute évidemment...

Reply:
Selon le fair-use américain et canadien, si tu es de Poly et que tu fais de la recherche avec les données Google: Fair use s'applique, tu fais pas mal ce que tu veux et tu peux publier aussi. Si tu es ARTM: ben ça dépend, est-ce d'intérêt public, est-ce des nouvelles journalistiques quand tu publies une analyse sur les déplacements dans une journal? Zone grise! Pour ça qu'il faut aller avec OpenStreetMap au plus vite!